src/main.js

```diff
@@ -94,4 +115,42 @@ export const addListener = (selector, callback) => {
   return elements;
 };
 
+function debounce(f, t) {
+  return function (args) {
+    let previousCall = this.lastCall;
+    this.lastCall = Date.now();
+    if (previousCall && this.lastCall - previousCall <= t) {
+      clearTimeout(this.lastCallTimer);
+    }
+    this.lastCallTimer = setTimeout(() => f(args), t);
+  };
+}
+
+function onSearch(e) {
```

Comment:
function on search should take query params and add to existing or replace keyword and put page to 1, after push state is called/ url is not needed

Reply:
fixed